src/main/java/gregtech/common/EventHandlers.java

```diff
@@ -206,9 +213,92 @@ public static void onLivingEquipmentChangeEvent(LivingEquipmentChangeEvent event
         }
     }
 
+    @SuppressWarnings({ "ConstantValue", "deprecation" })
     @SubscribeEvent
-    @SideOnly(Side.CLIENT)
     public static void onPlayerTick(TickEvent.PlayerTickEvent event) {
+        EntityPlayer player = event.player;
+        if (event.phase == TickEvent.Phase.START && !player.world.isRemote) {
+            IAttributeInstance movementSpeed = player.getEntityAttribute(SharedMonsterAttributes.MOVEMENT_SPEED);
+            if (movementSpeed == null) return;
+            AttributeModifier modifier = movementSpeed.getModifier(BlockUtility.WALKING_SPEED_UUID);
+
+            double speedBonus;
+            if (!player.onGround || player.isInWater() || player.isSneaking()) {
+                speedBonus = 0;
```

Comment:
why not just return here?

Reply:
Short circuiting here would cause the speed bonus to remain even after the player leaves the block.